go/mobile/async.go

```diff
@@ -0,0 +1,214 @@
+package iden3mobile
+
+import (
+	"encoding/json"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/iden3/go-iden3-core/db"
+	log "github.com/sirupsen/logrus"
+)
+
+type (
+	Event interface {
+		OnEvent(string, string, string, error) // Type, Ticket ID, Data(json), error
```

Comment:
Type aliases could be used here for more clarity.
Example:
```go
type TicketType string
const (
  TicketTypeA TicketType = "A"
  TicketTypeB TicketType = "B"
)
type TicketID string
func NewTicketID() TicketID {
  return randTicketID()
}
OnEvent(TicketType, TicketID, string, error)
```

Reply:
Note: I don't know how type aliases translate to java / objective-c.

Reply:
Will have a look at it, and if it maps well will use it.

Reply:
It seems that it doesn't work with gomobile.